pygmt/base_plotting.py

```diff
@@ -994,3 +994,358 @@ def text(
+            Ignored if `spec` is a DataFrame and contains a 'depth' column.
+        convention: str
+            ``"aki"`` (Aki & Richards), ``"gcmt"`` (global CMT), ``"mt"``
+            (seismic moment tensor), ``"partial"`` (partial focal mechanism),
+            or ``"principal_axis"`` (principal axis). Ignored if `spec` is a
+            dictionary or dataframe.
+        component: str
+            The component of the seismic moment tensor to plot. ``"full"`` (the
+            full seismic moment tensor), ``"dc"`` (the closest double couple
+            with zero trace and zero determinant), ``"deviatoric"`` (zero
+            trace)
+        plot_lon: int or float or list
+            Longitude(s) at which to place beachball, only used if `spec` is a
+            dictionary. List must be the length of the number of events.
+            Ignored if `spec` is a DataFrame and contains a 'plot_lon' column.
+        plot_lat: int or float or list
+            Latitude(s) at which to place beachball, only used if `spec` is a
+            dictionary. List must be the length of the number of events.
+            Ignored if `spec` is a DataFrame and contains a 'plot_lat' column.
+        offset: bool or str
+            Offsets beachballs to the longitude, latitude specified in
+            the last two columns of the input file or array, or by `plot_lon`
+            and `plot_lat` if provided. A small circle is plotted at the
+            initial location and a line connects the beachball to the circle.
+            Specify pen and optionally append ``+ssize`` to change the line
+            style and/or size of the circle.
+        {J}
+        {R}
+        {B}
+        """
+
+        # Check the spec and parse the data according to the specified
+        # convention
+        if isinstance(spec, (dict, pd.DataFrame)):
+            # dicts and DataFrames are handed similarly but not identically
+            if (lon is None or lat is None or depth is None) and not isinstance(
+                spec, pd.DataFrame
+            ):
+                raise GMTError("Location not fully specified.")
+
+            AKI_PARAMS = ["strike", "dip", "rake", "magnitude"]
+            GCMT_PARAMS = [
+                "strike1",
+                "dip1",
+                "rake1",
+                "strike2",
+                "dip2",
+                "rake2",
+                "mantissa",
+                "exponent",
+            ]
+            MT_PARAMS = ["mrr", "mtt", "mff", "mrt", "mrf", "mtf", "exponent"]
+            PARTIAL_PARAMS = ["strike1", "dip1", "strike2", "fault_type", "magnitude"]
+            PRINCIPAL_AXIS_PARAMS = [
+                "t_exponent",
+                "t_azimuth",
+                "t_plunge",
+                "n_exponent",
+                "n_azimuth",
+                "n_plunge",
+                "p_exponent",
+                "p_azimuth",
+                "p_plunge",
+                "exponent",
+            ]
```

Comment:
Could you put this into a dictionary? E.g. `{"AKI_PARAMS" : [...], "MT_PARAMS": [...], ...}`. This might help to simplify the if-then statements later.

Reply:
I sure can. Great idea.